packages/cerebral/src/Module.js

```diff
@@ -1,35 +1,24 @@
 class Module {
-  constructor (path, module, controller) {
-    this.name = path.slice().pop()
-    this.path = path
-    this.controller = controller
-    this.modules = {}
+  constructor (controller, path, moduleDescription) {
+    const moduleStub = {
+      controller,
+      path: path.join('.'),
+      name: path.slice().pop()
+    }
 
-    const moduleDescription = typeof module === 'function' ? module(this) : module
+    const module = typeof moduleDescription === 'function' ? moduleDescription(moduleStub) : moduleDescription
```

Comment:
Ah, so you did not want to define the module as a class... Just return class instance instead of object... okay, cool. You needed chnages for this? That should have been possible already? Anyways, nice cleanup! :)